### ISSUE TITLE
[BUG]: synchronized ^3.0.0-nullsafety

### ISSUE THREAD

**Issue by primemb** [Canardoux/tau#624]
## Flutter Sound Version : ^7.6.2+6

- **FULL** or **LITE** flavor ? Full

- **Important**: Result of the command : ```flutter pub deps | grep flutter_sound```

```
Because synchronized >=3.0.0-nullsafety.0 requires SDK version >=2.12.0-29 <3.0.0 and no versions of synchronized match >=3.0.0-nullsafety <3.0.0-nullsafety.0, synchronized >=3.0.0-nullsafety is forbidden.

So, because biomedian depends on flutter_sound ^7.6.2+6 which depends on synchronized ^3.0.0-nullsafety, version solving failed.
pub upgrade failed (1; So, because biomedian depends on flutter_sound ^7.6.2+6 which depends on synchronized ^3.0.0-nullsafety, version solving failed.)
exit code 1
```

Flutter doctor:

```
All done!
[√] Flutter: is fully installed. (Channel stable, 1.22.6, on Microsoft Windows
    [Version 10.0.19041.746], locale en-US)
[!] Android toolchain - develop for Android devices: is partially installed;
    more components are available. (Android SDK version 29.0.3)
[√] Android Studio: is fully installed. (version 4.0)
[√] Connected device: is fully installed. (1 available)
```

my pubspec.yaml :

```
environment:
  sdk: ">=2.7.0 <3.0.0"

dependencies:
  flutter:
    sdk: flutter

  # The following adds the Cupertino Icons font to your application.
  # Use with the CupertinoIcons class for iOS style icons.
  cupertino_icons: ^0.1.2
  http: ^0.12.0+2
  modal_progress_hud: ^0.1.3
  shared_preferences: ^0.5.6+2
  badges: ^1.1.0
  flutter_map: ^0.8.2
  latlong: ^0.6.1
  url_launcher: ^5.4.2
  cached_network_image: ^2.0.0
  flutter_sim_country_code: ^0.0.3
  country_code_picker: ^1.5.0
  country_codes: ^1.0.1
  provider: ^4.0.4
  charts_flutter: ^0.9.0
  webview_flutter: ^0.3.22+1
  persian_number_utility: ^0.1.0
  flutter_phone_direct_caller: ^1.0.1
  touchable: ^0.2.0
  font_awesome_flutter: ^8.11.0
  flutter_cached_pdfview: ^0.3.4
  image_picker: ^0.6.7+21
  flutter_sound: ^7.6.2+6
  permission_handler: ^5.0.1+1
  shamsi_date: ^0.9.1
  fluttertoast: ^7.1.6
  mime: ^0.9.7
  http_parser: ^3.1.4
  easy_localization: ^3.0.0-dev.1

dev_dependencies:
  flutter_test:
    sdk: flutter

dependency_overrides:
  intl: ^0.16.1
  path: 1.7.0
```

**Comment by Larpoux** [comment]
Hi Mohammad,

Dart/Flutter are doing a complete mess with NullSafety.
The actual situation of Dart is really bad : NullSafety is incompatible with the old behavior of the compiler.

I do not remember why I had to upgrade our plugin `synchronized ^3.0.0-nullsafety` but this is an error and I am going to fix that.

Someone (me?) will have upgrade Flutter Sound and use the NullSafety option of the compiler, but this upgrade must be a parallel branch, and not the master branch.

[#584] is also something about NullSafety

**Comment by MATTYGILO** [comment]
Has the branch been made?

**Comment by Larpoux** [comment]
I am currently doing a new release without any dependency to synchronized ^3.0.0-nullsafety.
Please when one hour.

**Comment by MATTYGILO** [comment]
@Larpoux No worries, Thanks anyway

**Comment by Larpoux** [comment]
Fixed in Flutter Sound release 7.6.3
